src/redisearch_rs/inverted_index/src/index_result.rs

```diff
@@ -203,6 +237,42 @@ impl<'index> RSTermRecord<'index> {
             unsafe { std::slice::from_raw_parts(offsets.data as *const u8, offsets.len as usize) }
         }
     }
+
+    /// Get the query term pointer of this term record.
+    pub fn query_term(&self) -> *mut RSQueryTerm {
+        match self {
+            RSTermRecord::Borrowed { term, .. } => *term,
+            RSTermRecord::Owned { term, .. } => *term,
+        }
+    }
+
+    /// Create a deep copy of this term record, allocating new memory for the offsets, but reusing the term.
+    pub fn deep_copy(&self) -> RSTermRecord<'static> {
```

Comment:
Why not use `ToOwned` instead?

Reply:
I forgot about it :sweat_smile: . But it is also giving a compile error
<img width="853" height="344" alt="image" src="https://github.com/user-attachments/assets/6e4e1f2e-7d51-4e47-8990-25b0af3d0422" />

Reply:
I remember the same. Let me check whether I have this impl wrong...

Reply:
There's an implementation conflict if your type implements `Clone`. You need to use an inherent method (e.g. see `Cow::into_owned` as a reference).

Reply:
Renamed this to `into_owned` to align with `Cow`

https://github.com/RediSearch/RediSearch/pull/6554/commits/0fd3c0658b9cf835cc77b0ba553662126039e577